packages/react-native-windows-init/src/Cli.ts

```diff
@@ -0,0 +1,291 @@
+ *
+ * @format
+ */
+
+import * as yargs from 'yargs';
+import * as fs from 'fs';
+import * as semver from 'semver';
+import {execSync} from 'child_process';
+import * as validUrl from 'valid-url';
+import * as prompts from 'prompts';
+import * as findUp from 'find-up';
+import * as chalk from 'chalk';
+
+const Registry = require('npm-registry');
+
+let npmConfReg = execSync('npm config get registry')
+  .toString()
+  .trim();
+let NPM_REGISTRY_URL = validUrl.isUri(npmConfReg)
+  ? npmConfReg
+  : 'http://registry.npmjs.org';
+const npm = new Registry({registry: NPM_REGISTRY_URL});
+
+const argv = yargs.options({
+  version: {
+    type: 'string',
+    describe: 'The version of react-native-windows to use.',
+  },
+  namespace: {type: 'string', describe: 'The native project namespace.'},
+  verbose: {type: 'boolean', describe: 'Enables logging.'},
+  language: {
+    type: 'string',
+    describe: 'Which language the app is written in.',
+    choices: ['cs', 'cpp'],
+    default: 'cpp',
+  },
+  overwrite: {
+    type: 'boolean',
+    describe: 'Overwrite any existing files without prompting',
+  },
+}).argv;
+
+const EXITCODE_NO_MATCHING_RNW = 2;
+const EXITCODE_UNSUPPORTED_VERION_RN = 3;
+const EXITCODE_USER_CANCEL = 4;
+
+function REACT_NATIVE_WINDOWS_GENERATE_PATH() {
+  return require.resolve('react-native-windows/local-cli/generate-windows.js', {
+    paths: [process.cwd()],
+  });
+}
+
+function getReactNativeAppName() {
+  console.log('Reading application name from package.json...');
+  let name = JSON.parse(fs.readFileSync('package.json', 'utf8')).name;
+  if (!name) {
+    if (fs.existsSync('app.json')) {
+      console.log('Reading application name from app.json...');
+      name = JSON.parse(fs.readFileSync('app.json', 'utf8')).name;
+    }
+  }
+  if (!name) {
+    console.error('Please specify name in package.json or app.json');
+  }
+  return name;
+}
+
+function getReactNativeVersion() {
+  console.log('Reading react-native version from node_modules...');
+  const rnPkgJsonPath = require.resolve('react-native/package.json', {
+    paths: [process.cwd()],
+  });
+  if (fs.existsSync(rnPkgJsonPath)) {
+    return require(rnPkgJsonPath).version;
+  }
+
+  console.error(
+    'Error: Must be run from a project that already depends on react-native, and has react-native installed.',
+  );
+  process.exit(1);
+}
+
+function ErrorOutOnUnsupportedVersionOfReactNative(rnVersion: string) {
+  console.error(`Error: Unsupported version of react-native: ${chalk.cyan(
+    rnVersion,
+  )}
+react-native-windows supports react-native versions ${chalk.cyan('>=0.60')}`);
+  process.exit(EXITCODE_UNSUPPORTED_VERION_RN);
+}
+
+function getDefaultReactNativeWindowsSemVerForReactNativeVersion(
+  rnVersion: string,
+) {
+  const validVersion = semver.valid(rnVersion);
+  if (validVersion) {
+    const major = semver.major(validVersion);
+    const minor = semver.minor(validVersion);
+    if (major === 0 && minor >= 60) {
```

Comment:
Should we include 59 as supported?

Reply:
I dont think anyone should be using 59 at this point. -- But I guess it doesn't hurt.  It'll find the -vnext version and give them a prerelease warning at least.